packages/compiler/test/server/get-hover.test.ts

```diff
@@ -0,0 +1,374 @@
+import { deepStrictEqual } from "assert";
+import { describe, it } from "vitest";
+import { Hover, MarkupKind } from "vscode-languageserver/node.js";
+import { createTestServerHost, extractCursor } from "../../src/testing/test-server-host.js";
+
+describe("compiler: server: on hover", () => {
+  describe("get hover for scalar", () => {
+    it("string scalar", async () => {
+      const hover = await getHoverAtCursor(
+        `
+          scalar Str┆ing;
+        `
+      );
+      deepStrictEqual(hover, {
+        contents: {
+          kind: MarkupKind.Markdown,
+          value: "```typespec\n" + "scalar String\n" + "```",
+        },
+      });
+    });
+
+    it("int16 scalar", async () => {
```

Comment:
i would change those test to be 
- `hover scalar declaration`
- `hover scalar reference` (where this is testing soemthing like `scalar foo is mySt┆ring;

Reply:
sure, but seems scalar doesnt support "is", so use "extends" in the test